### ISSUE TITLE
Crash in BugsnagSystemState.m

### ISSUE THREAD

**Issue by sethfri** [bugsnag/bugsnag-cocoa#886]
### Describe the bug
We're seeing a crash of unknown origin that seems to be coming from Bugsnag. We have only seen this crash appear 2 times, affecting 2 customers, so it's pretty rare. I also realize the stack trace isn't particularly helpful, so please let me know if you need info from other threads in the report.

Crashlytics reports this as
```
Crashed: com.apple.root.default-qos
SIGSEGV 0x0000000af24b6e20
```

Bugsnag reports this as
```
EXC_BAD_ACCESS_objc_retain
Attempted to dereference garbage pointer 0xaf24b6e20.
```

### Steps to reproduce
Unknown

### Environment
* Bugsnag version: 6.2.3
* CocoaPods version: 1.9.3
* Carthage version (if any): n/a
* iOS/tvOS/macOS version(s): 13.7.0
* Simulator or physical device: iPhone 6s
* XCode version: 12.1
* Swift version (if applicable): 5.1

### Stack Trace

<details><summary>Stack trace:</summary>

```
Crashed: com.apple.root.default-qos
0  libobjc.A.dylib                0x18c299980 objc_retain + 16
1  Cash                           0x10184a52c -[BugsnagSystemState sync] + 203 (BugsnagSystemState.m:203)
2  libdispatch.dylib              0x18c2219a8 _dispatch_call_block_and_release + 24
3  libdispatch.dylib              0x18c222524 _dispatch_client_callout + 16
4  libdispatch.dylib              0x18c1cab54 _dispatch_queue_override_invoke + 684
5  libdispatch.dylib              0x18c1d7438 _dispatch_root_queue_drain + 376
6  libdispatch.dylib              0x18c1d7bf8 _dispatch_worker_thread2 + 124
7  libsystem_pthread.dylib        0x18c273b38 _pthread_wqthread + 212
8  libsystem_pthread.dylib        0x18c276740 start_wqthread + 8
```
</details>

**Comment by johnkiely1** [comment]
Hi @sethfri,

Thanks for raising this, we will take a look at get back to you when we know more.

**Comment by nickdowell** [comment]
Hi @sethfri if you could let us know the version of your app this crash occurred in, it would help us - thanks!

**Comment by johnkiely1** [comment]
Hi @sethfri.

Just to let you know we have identified and fixed the issue causing this. The fix has been released in [v6.2.5.](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.2.5).

Thanks again for raising this. We appreciate it.
